### PR TITLE
Mask unwanted neutron services for compute hosts

### PR DESCRIPTION
**Pull request by wangxian-ibm**
Only neutron-linuxbridge-agent service is needed on compute hosts however, all neutron services are installed. When running `neutron-reset` on compute hosts, these unwanted services will get started.

### CODE REVIEW

**Comment by nirajdp76** [comment]
yep @paulczar comment makes sense. I will check one other place where this was done as well and get it corrected.

File: roles/neutron-data/tasks/main.yml [hunk 1]
```diff
@@ -49,6 +49,18 @@
...
+- name: mask unwanted neutron services on compute (rhel)
+  command: systemctl mask {{ item.name }}
+  with_items:
+    - "{{ neutron.services.neutron_server }}"
+    - "{{ neutron.services.neutron_dhcp_agent }}"
+    - "{{ neutron.services.neutron_l3_agent }}"
+    - "{{ neutron.services.neutron_metadata_agent }}"
+  when:
+    - ursula_os == 'rhel'
+    - inventory_hostname in groups['compute']
```
Comment (paulczar) [inline roles/neutron-data/tasks/main.yml]: why is this needed if the following check already matches ?

Comment (wangxian-ibm) [inline roles/neutron-data/tasks/main.yml]: Because `neutron-data` role only runs for `compute:network` hosts, we could get rid of this check but do we want to rely on that and not be strict at checking?

Comment (nirajdp76) [inline roles/neutron-data/tasks/main.yml]: @wangxian-ibm what will happen if inventory_hostname is also in groups[controller] ?

Comment (wangxian-ibm) [inline roles/neutron-data/tasks/main.yml]: `inventory_hostname` is never in `groups[controller]` because `neutron-data` only runs on compute and network hosts. In that case, we can just check for not `network` hosts.
```
  when:
    - ursula_os == 'rhel'
    - not inventory_hostname in groups['network']
```
Bottomline is that we need to consider the converged controllers scenario. However checking for just not `network` hosts will allow us to separate `network` nodes from `controller` nodes easily in the future.

Comment (nirajdp76) [inline roles/neutron-data/tasks/main.yml]: right, current when condition will also mask neutron services on controllers (which are part of network group), right?
So we should be changing when as u mentioned in ur comment.

**Comment by wangxian-ibm** [comment]
@nirajdp76 @paulczar PR updated and tested

**Comment by nirajdp76** [comment]
:+1:

**Comment by masteinhauser** [comment]
retest

**Comment by nirajdp76** [comment]
retest

**Comment by nirajdp76** [comment]
retest

**Comment by ddaskal** [comment]
retest

**Comment by wangxian-ibm** [comment]
retest
